### PR TITLE
Fixed changing configuration name when adding new device

### PR DESCRIPTION
**Pull request by mchf**
bnc#912904

### CODE REVIEW

File: src/modules/LanItems.rb [hunk 1]
```diff
@@ -1932,45 +1934,43 @@ def Commit
...
+          newdev["BONDING_SLAVE#{i}"] = slave
+          i += 1
         end
 
         #assign nil to rest BONDING_SLAVEn to remove them
-        while Ops.less_than(i, @MAX_BOND_SLAVE)
-          Ops.set(newdev, Builtins.sformat("BONDING_SLAVE%1", i), nil)
-          i = Ops.add(i, 1)
+        while i < @MAX_BOND_SLAVE
+          newdev["BONDING_SLAVE#{i}"]
```
Comment (jreidinger) [inline src/modules/LanItems.rb]: ugh, how this should work?
```
irb(main):001:0> h = {}
=> {}
irb(main):002:0> h["binding0"] = true
=> true
irb(main):003:0> h
=> {"binding0"=>true}
irb(main):004:0> h["binding1"]
=> nil
irb(main):005:0> h
=> {"binding0"=>true}
```

File: src/modules/LanItems.rb [hunk 2]
```diff
@@ -1932,45 +1934,43 @@ def Commit
...
-        Ops.set(newdev, "WIRELESS_KEY", "") # TODO: delete the varlable
-        Ops.set(newdev, "WIRELESS_KEY_0", Ops.get(@wl_key, 0, ""))
-        Ops.set(newdev, "WIRELESS_KEY_1", Ops.get(@wl_key, 1, ""))
-        Ops.set(newdev, "WIRELESS_KEY_2", Ops.get(@wl_key, 2, ""))
-        Ops.set(newdev, "WIRELESS_KEY_3", Ops.get(@wl_key, 3, ""))
+        newdev["WIRELESS_KEY"] = "" # TODO: delete the varlable
+        newdev["WIRELESS_KEY_0"] = Ops.get(@wl_key, 0, "")
+        newdev["WIRELESS_KEY_1"] = Ops.get(@wl_key, 1, "")
+        newdev["WIRELESS_KEY_2"] = Ops.get(@wl_key, 2, "")
+        newdev["WIRELESS_KEY_3"] = Ops.get(@wl_key, 3, "")
```
Comment (jreidinger) [inline src/modules/LanItems.rb]: NP: I think here is better to use merge approach:
```
newdev.merge! ({
  WIRELESS_MODE" => @wl_mode,
  ....
)}
```

**Comment by jreidinger** [comment]
one new bug, one nitpick and no tests. Please fix bug and also add some tests that proof that you do not break it ;)

**Comment by mchf** [comment]
there are tests already which covers bugfixed code.

**Comment by jreidinger** [comment]
LGTM

File: src/modules/LanItems.rb [hunk 3]
```diff
@@ -540,8 +540,10 @@ def ReplaceItemUdev(replace_key, new_key, new_val)
     # Returns new name
     def SetItemName( itemId, name)
       if name && !name.empty?
-        updated_rule = update_udev_rule_key(GetItemUdevRule(itemId), "NAME", name)
-        @Items[itemId]["udev"]["net"] = updated_rule
+        if @Items[itemId]["udev"]
```
Comment (mvidner) [inline src/modules/LanItems.rb]: Shouldn't we instead ensure that `@Items[itemId]` has a `udev` key whose value is a hash?

File: src/modules/LanItems.rb [hunk 4]
```diff
@@ -1932,45 +1934,43 @@ def Commit
       case @type
       when "bond"
         i = 0
-        Builtins.foreach(@bond_slaves) do |slave|
-          Ops.set(newdev, Builtins.sformat("BONDING_SLAVE%1", i), slave)
-          i = Ops.add(i, 1)
+        @bond_slaves.each do |slave|
```
Comment (mvidner) [inline src/modules/LanItems.rb]: Use `Enumerable#each_with_index`

File: src/modules/LanItems.rb [hunk 5]
```diff
@@ -1932,45 +1934,43 @@ def Commit
...
+          newdev["BONDING_SLAVE#{i}"] = slave
+          i += 1
         end
 
         #assign nil to rest BONDING_SLAVEn to remove them
-        while Ops.less_than(i, @MAX_BOND_SLAVE)
-          Ops.set(newdev, Builtins.sformat("BONDING_SLAVE%1", i), nil)
-          i = Ops.add(i, 1)
+        while i < @MAX_BOND_SLAVE
```
Comment (mvidner) [inline src/modules/LanItems.rb]: Ah, never mind, `i` is still needed here.

**Comment by jreidinger** [comment]
OK, LGTM, I think this part of code deserve bigger refactoring, it is quite messy

**Comment by mchf** [comment]
@jreidinger: 
Yes, this part of yast-network wasn't touched that much ... i know that this is small step for you ... but huge step for the code (hopefully) ;-)

Thanks for reviews

**Pull request merged**